libwavesync/packetizer.py

```diff
@@ -38,6 +39,19 @@ def create_socket(self, channels, ttl, multicast_loop, broadcast):
                              socket.IP_MULTICAST_TTL,
                              ttl)
 
+        for address, port in channels:
+            if source_address and ipaddress.IPv4Address(address).is_multicast:
+                try:
+                    self.sock.setsockopt(socket.SOL_IP,
+                                         socket.IP_MULTICAST_IF,
+                                         socket.inet_aton(source_address))
+                    print("added membership, interface source address: %s, group: %s" % (source_address, address))
```

Comment:
Would probably capitalize the messages and revert the condition to decrease the indent (continue on not multicast). The except is also way to generic - but the change is worthy. ;-) Thank you for that and said changes can be fixed later.